app/controllers/lesson_completions_controller.rb

```diff
@@ -1,51 +1,43 @@
 class LessonCompletionsController < ApplicationController
   
-  before_filter :authenticate_request
+  before_action :authenticate_request
+  before_action :lookup_lesson
   
   def create
-    # Validate that the id does in fact correspond to a real lesson
-    # Note that the authentication before_filter makes sure we've got a current_user
-    @lesson = Lesson.where(:id => params[:lesson_id]).first
-
-    if @lesson.nil?
-      render :nothing => true, :status => 400 # bad request
-    else   
-      @course = @lesson.course
-      @next_lesson = @lesson.next_lesson
-      lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
-      if lc.save
-        render "create", :formats => [:js]
-      else
-        render :nothing => true, :status => 400 # bad request
-      end
+    @course = @lesson.course
+    @next_lesson = @lesson.next_lesson
+    lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
+    if lc.save
+      render :create, formats: [:js]
+    else
+      head :bad_request
     end
   end
   
   def destroy
-    # Validate that the id does in fact correspond to a real lessons
-    # Note that the authentication before_filter makes sure we've got a current_user
-    @lesson = Lesson.where(:id => params[:lesson_id]).first
-
-    if @lesson.nil?
-      render :nothing => true, :status => 400 # bad request
-    else   
-      @course = @lesson.course
-      @next_lesson = @lesson.next_lesson
-      lc = LessonCompletion.where(:student_id => current_user.id, :lesson_id => @lesson.id).first
-      if lc.nil? 
-        render :nothing => true, :status => 400 # bad request
-      elsif lc.delete
-        render "create", :formats => [:js] 
-      else
-        render :nothing => true, :status => 400 # bad request
-      end
+    @course = @lesson.course
+    @next_lesson = @lesson.next_lesson
+    lc = LessonCompletion.where(:student_id => current_user.id, :lesson_id => @lesson.id).
```

Comment:
I don't think we need this, can we remove it please? and just call `@lesson.destroy` at the bottom instead